src/js/components/stock-movement-wizard/inbound/AddItemsPage.jsx

```diff
@@ -232,8 +233,40 @@ class AddItemsPage extends Component {
    * @public
    */
   getLineItemsToBeSaved(lineItems) {
-    const lineItemsToBeAdded = _.filter(lineItems, item => !item.id);
-    const lineItemsToBeUpdated = _.filter(lineItems, item => item.id);
+    const lineItemsToBeAdded = _.filter(lineItems, item =>
+      !item.statusCode && item.quantityRequested && item.quantityRequested !== '0' && item.product);
+    const lineItemsWithStatus = _.filter(lineItems, item => item.statusCode);
+    const lineItemsToBeUpdated = [];
+    _.forEach(lineItemsWithStatus, (item) => {
+      const oldItem = _.find(this.state.currentLineItems, old => old.id === item.id);
+      const oldQty = parseInt(oldItem.quantityRequested, 10);
+      const newQty = parseInt(item.quantityRequested, 10);
+      const oldRecipient = oldItem.recipient && _.isObject(oldItem.recipient) ?
+        oldItem.recipient.id : oldItem.recipient;
+      const newRecipient = item.recipient && _.isObject(item.recipient) ?
+        item.recipient.id : item.recipient;
+
+      // Intersection of keys common to both objects (excluding product key)
+      const keyIntersection = _.remove(
+        _.intersection(
+          _.keys(oldItem),
+          _.keys(item),
+        ),
+        key => key !== 'product',
+      );
+
+      if (
+        (this.state.values.origin.type === 'SUPPLIER' || !this.state.values.hasManageInventory) &&
```

Comment:
Do we need this condition? Inbound is always supplier or other origin without manage inventory.